lib/t_web/channels/call_channel.ex

```diff
@@ -0,0 +1,68 @@
+defmodule TWeb.CallChannel do
+  use TWeb, :channel
+
+  alias TWeb.Presence
+  alias T.Calls
+
+  @impl true
+  def join("call:" <> call_id, _params, socket) do
+    %{current_user: current_user, screen_width: screen_width} = socket.assigns
+    socket = assign(socket, call_id: call_id)
+
+    case Calls.get_call_role_and_peer(call_id, current_user.id) do
+      {:ok, :caller = role, _peer} ->
+        send(self(), :after_join)
+        reply = %{ice_servers: Calls.ice_servers()}
+        {:ok, reply, assign(socket, role: role)}
+
+      {:ok, :called = role, peer} ->
+        send(self(), :after_join)
+        reply = %{caller: render_peer(peer, screen_width), ice_servers: Calls.ice_servers()}
+        {:ok, reply, assign(socket, role: role)}
+
+      {:error, :not_found} ->
+        {:error, %{"reason" => "not_found"}}
+
+      {:error, :ended} ->
+        {:error, %{"reason" => "ended"}}
+    end
+  end
+
+  @impl true
+  def handle_in("peer-message", %{"body" => body}, socket) do
+    me = socket.assigns.current_user.id
+    # TODO check that the peer is online?
```

Comment:
TODO

Reply:
not sure how to handle right now, maybe a buffer of peer messages on the channel, or maybe a buffer of peer messages on the client